packages/realm-react/src/types.ts

```diff
@@ -0,0 +1,93 @@
+////////////////////////////////////////////////////////////////////////////
+//
+// Copyright 2023 Realm Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+////////////////////////////////////////////////////////////////////////////
+
+export enum AuthOperationName {
+  LogIn = "logIn",
+  LogInWithAnonymous = "logInWithAnonymous",
+  LogInWithApiKey = "logInWithApiKey",
+  LogInWithEmailPassword = "logInWithEmailPassword",
+  LogInWithJWT = "logInWithJWT",
+  LogInWithGoogle = "logInWithGoogle",
+  LogInWithApple = "logInWithApple",
+  LogInWithFacebook = "logInWithFacebook",
+  LogInWithFunction = "logInWithFunction",
+  LogOut = "logOut",
+  Register = "register",
+  Confirm = "confirm",
+  ResendConfirmationEmail = "resendConfirmationEmail",
+  RetryCustomConfirmation = "retryCustomConfirmation",
+  SendResetPasswordEmail = "sendResetPasswordEmail",
+  ResetPassword = "resetPassword",
+  CallResetPasswordFunction = "callResetPasswordFunction",
+  None = "none",
```

Comment:
Wouldn't it make more sense to represent this as `undefined` or `null`?

```suggestion
```

Reply:
I could live with `null`